### DISCUSSION TITLE
Bad performance with old, migrated database schema

### DISCUSSION THREAD

**Discussion by spupuz** [photoprism/photoprism#1067] [Q&A]
Linux Ubuntu 20.10 client Mozilla latest nightly as browser, or chrome on android 10 Samsung s10.

Host OS Debian OpenMediaVault Linux 
docker preview image 
CPU i7-5775c @3.30ghz 
RAM 32 GB  
disks for photos ext4 mirror of 2 TB WD RedTab
disk for database ZFS Raid 6 of 6 disks 1 TB + 1 SSD 256 cache

library dimension 35000 original images 3000 videos

While scrolling the images with the smallest preview thumbnail the every X second the scroll stop for 2/3 seconds for loading the images.

**Comment by lastzero** [comment]
Did you test with a fresh database already as discussed in the chat?

**Comment by spupuz** [comment]
What I'm supposed to rename?
This is my MySQL DB folder:

```
root@openmediavault:/zfsfiles/DockerConfig/Photoprism/mysql# ls -lai
total 121959
652735 drwxr-xr-x 5 openmediavault-webgui root                         12 Feb  9 04:03 .
155764 drwxrwxrwx 7 root                  root                          8 Feb  9 04:03 ..
653863 -rw-rw---- 1 openmediavault-webgui openmediavault-config    139264 Feb  9 09:27 aria_log.00000001
653864 -rw-rw---- 1 openmediavault-webgui openmediavault-config        52 Feb  9 09:27 aria_log_control
104018 -rw-rw---- 1 openmediavault-webgui openmediavault-config     13500 Feb  9 04:03 ib_buffer_pool
655343 -rw-rw---- 1 openmediavault-webgui openmediavault-config  79691776 Feb  9 04:03 ibdata1
653866 -rw-rw---- 1 openmediavault-webgui openmediavault-config 100663296 Feb  9 08:48 ib_logfile0
104020 -rw-rw---- 1 openmediavault-webgui openmediavault-config  12582912 Feb  9 04:03 ibtmp1
658089 -rw-rw---- 1 openmediavault-webgui openmediavault-config         0 Dec  6 14:45 multi-master.info
652736 drwx------ 2 openmediavault-webgui openmediavault-config        90 Dec  6 14:45 mysql
652737 drwx------ 2 openmediavault-webgui openmediavault-config         3 Dec  6 14:45 performance_schema
652738 drwx------ 2 openmediavault-webgui openmediavault-config        53 Dec 31 16:06 photoprism
root@openmediavault:/zfsfiles/DockerConfig/Photoprism/mysql# 
```

**Comment by lastzero** [comment]
Start with a fresh database. For example, you may stop the mariadb service, change its storage path to a different local directory (don't change the path inside the container) and then start it again (docker-compose up). This will keep your existing database files but starts the database service with an empty scheme using the same settings as before so that you don't have to change other config variables.

**Comment by spupuz** [comment]
Ok I'll try that, does it need to re-index everything I suppose?

**Comment by spupuz** [comment]
```
photoprism@dd171bbccd89:~$ photoprism cleanup
INFO[2021-02-09T09:30:37Z] cleanup: removed 5408 index entries and 5452 orphan thumbnails in 21m31.116839254s 
INFO[2021-02-09T09:30:37Z] closed database connection   
```
I have made a cleanup and I'm re-indexing then I'll move to a new DB, indexing 30k photos is not so fast

**Comment by lastzero** [comment]
Is performance getting better with a fresh database?

**Reply by spupuz** [reply]
I have modified the MySQL folder today (I was busy previously) and I'm re-indexing I'll let you know as soon the procedure has finished.

**Comment by spupuz** [comment]
Hi @lastzero I recreated MySQL and re-indexed everything but this has not given major improvement in performances

**Reply by lastzero** [reply]
Was worth a try... can you record a screen video that shows the specific issues? Possible you're using the app in a way we didn't anticipate, or it's a transport layer / client issue.

Need to figure out if this needs action now (low hanging fruit?) or will be solved with our continuous frontend optimizations anyway.
